### PR TITLE
Rewrite public feature specs

### PR DESCRIPTION
**Pull request by jamiecobbett**
Use a style that avoids too much implementation detail in the specs and tries to be clearer and more expressive.

### CODE REVIEW

File: spec/features/steps/public/contact_steps.rb [hunk 1]
```diff
@@ -2,29 +2,34 @@ module Public
...
-        within(contacts_list_selector) do
-          contacts.each do |contact|
-            has_content? contact.title
-          end
-        end
-    end
-
     def should_list_contact(contact)
-      expect(page).to have_content(contact.title)
+       expect(page).to have_link(contact.title, :href => "/government/organisations/#{hmrc.slug}/contact/#{contact.slug}")
```
Comment (fatbusinessman) [inline spec/features/steps/public/contact_steps.rb]: Minor nit-pick: that seems to have become a 3-space indent.

**Comment by fatbusinessman** [comment]
One apparent typo, but apart from that, this looks very nice. Approved :+1:

File: spec/features/steps/public/contact_steps.rb (hunk 1 above)
Comment (jamiecobbett) [inline spec/features/steps/public/contact_steps.rb]: Rebased :space_invader:

**Comment by alext** [comment]
Looks good to me, but needs a rebase.

**Comment by jamiecobbett** [comment]
Rebased

**Pull request merged**